services/eventservice/src/tests.rs

```diff
@@ -0,0 +1,508 @@
+    let manager = data_manager::EventManager::new(&"./test/geteventstestdb".to_string());
+    let cat1 = Category {
+        name: "first".to_string(),
+    };
+    let cat2 = Category {
+        name: "second".to_string(),
+    };
+    let event1 = types::Event {
+        id: 1,
+        title: "Event1".to_string(),
+        cover: "N/A".to_string(),
+        description: "Empty".to_string(),
+        organizer: 1,
+        date_created: chrono::Utc::now().timestamp(),
+        date_planning: (chrono::Utc::now() + chrono::Duration::days(3)).timestamp(),
+        category: cat1.clone(),
+    };
+    let event2 = types::Event {
+        id: 2,
+        title: "Event2".to_string(),
+        cover: "N/A".to_string(),
+        description: "Empty".to_string(),
+        organizer: 1,
+        date_created: chrono::Utc::now().timestamp(),
+        date_planning: (chrono::Utc::now() + chrono::Duration::days(2)).timestamp(),
+        category: cat2.clone(),
+    };
+    let event3 = types::Event {
+        id: 3,
+        title: "Event3".to_string(),
+        cover: "N/A".to_string(),
+        description: "Empty".to_string(),
+        organizer: 2,
+        date_created: chrono::Utc::now().timestamp() + 10,
+        date_planning: (chrono::Utc::now() + chrono::Duration::days(1)).timestamp(),
+        category: cat1.clone(),
+    };
+    manager
+        .create_category(&cat1)
+        .expect("Failed to create category!");
+    manager
+        .create_category(&cat2)
+        .expect("Failed to create category!");
+    manager
+        .create_event(&event1)
+        .expect("Failed to create event");
+    manager
+        .create_event(&event2)
+        .expect("Failed to create event");
+    manager
+        .create_event(&event3)
+        .expect("Failed to create event");
+    assert_eq!(
+        manager
+            .get_events(Some(&"title".to_string()), None, None)
+            .unwrap(),
+        vec![event1.clone(), event2.clone(), event3.clone()]
+    );
+    assert_eq!(
+        manager
+            .get_events(Some(&"date_planning".to_string()), None, None)
+            .unwrap(),
+        vec![event3.clone(), event2.clone(), event1.clone()]
+    );
+    assert_eq!(
+        manager
+            .get_events(Some(&"title".to_string()), Some(1), None)
+            .unwrap(),
+        vec![event1.clone(), event2.clone()]
+    );
+    assert_eq!(
+        manager
+            .get_events(Some(&"title".to_string()), None, Some(&cat2.name))
+            .unwrap(),
+        vec![event2.clone()]
+    );
+    assert_eq!(
+        manager
+            .get_events(Some(&"title".to_string()), Some(1), Some(&cat1.name))
+            .unwrap(),
+        vec![event1.clone()]
+    );
+    manager._reset_all().expect("Failed to delete db");
+}
+
+#[test]
+fn category_get_delete_test() {
+    let manager = data_manager::EventManager::new(&"./test/cattestdb".to_string());
+    let cat1 = Category {
+        name: "first".to_string(),
```

Comment:
owww. Can't you simply create a string in rust?

Reply:
you can write 

- String::from("abc")
- "abc".to_string()
- "abc".to_owned()
- And even "abc".into()

The last variant may be simpler, but it is slower than other three (which are near the same in productivity)